File: bench/complex/5.cpp
#include <iostream>
#include <complex>
#include <ctime>
#include <cstdlib>
#include <chrono>

#define X 99999999.0

using namespace std;

float rand_float()
{
	return static_cast<float>(rand()) / (static_cast<float>(RAND_MAX / X));
}

int main(int argc, char const *argv[])
{
	srand(time(NULL));

	int n = 50000000; // Number of operations
	if (argc == 2)
	{
		n = atoi(argv[1]);
	}

	auto start = chrono::steady_clock::now();

	for (int i = 0; i < n; i++)
	{
		complex<float> comp(rand_float(), rand_float());
		complex<float> c = cos(comp);
		complex<float> s = sin(comp);
	}

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}

File: bench/complex/5.go
package main

import (
	"math/cmplx"
	"math/rand"
	"os"
	"strconv"
	"fmt"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())

	var n = 50000000 // Number of operations
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.Atoi(args[0])
	}

	var start = time.Now()

	for i := 0; i < n; i++ {
		comp := complex128(complex(rand.Float32(), rand.Float32()))
		_ = cmplx.Cos(comp)
		_ = cmplx.Sin(comp)
	}

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}
